src/NLog/Internal/PropertyHelper.cs

```diff
@@ -393,6 +391,27 @@ private static bool TryCreateCollectionObject(Type collectionType, string valueR
             return false;
         }
 
+        /// <summary>
+        /// Attempt to reuse the HashSet.Comparer from the original HashSet-object (Ex. StringComparer.OrdinalIgnoreCase)
+        /// </summary>
+        private static object[] CreateHashSetConstructorArgs(object obj, PropertyInfo propInfo)
+        {
+            var exitingValue = propInfo.CanRead && propInfo.GetIndexParameters().Length == 0 && propInfo.GetGetMethod() != null ? propInfo.GetGetMethod().Invoke(obj, null) : null;
+            if (exitingValue != null)
+            {
+                // Found original HashSet-object. See if we can extract the Comparer
+                var comparerPropInfo = exitingValue.GetType().GetProperty("Comparer", BindingFlags.Instance | BindingFlags.Public);
+                if (comparerPropInfo != null && comparerPropInfo.CanRead && comparerPropInfo.GetGetMethod() != null)
+                {
+                    var comparer = comparerPropInfo.GetGetMethod().Invoke(exitingValue, null);
```

Comment:
Why not `comparerPropInfo.GetValue(existing Value)`? 

(Also above like this)

Reply:
Fixed